.github/workflows/test.yml

```diff
@@ -40,14 +45,30 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
+
+    - name: Set env and whether or not to use a local server
+      id: set-env-step
+      run: |
+        if ${{ github.event_name == 'pull_request' || github.event_name == 'push' }}; then
+          ENV=local
+          LOCAL_SERVER=${{ matrix.gradleTask == 'integrationTest' }}
+        elif ${{ github.event_name == 'workflow_dispatch' }}; then
+          ENV=${{ github.event.inputs.testEnv }}
+          LOCAL_SERVER=false
+        fi
```

Comment:
should this have a case for
```
else
  echo ::error ::${{ github.event_name }} not supported for this workflow
  exit 1
```
?  I guess we don't define any other events in the `on:` section,  so it's not necessary, your  call on this.